youtube_dl/YoutubeDL.py

```diff
@@ -1208,6 +1208,8 @@ def process_video_result(self, info_dict, download=True):
             raise ExtractorError('Missing "id" field in extractor result')
         if 'title' not in info_dict:
             raise ExtractorError('Missing "title" field in extractor result')
+        if not info_dict.get('formats'):
```

Comment:
Note that the `formats` field is optional, the `url` can be set in the main dict. For example the appleconnect test URL (https://itunes.apple.com/us/post/idsa.4ab17a39-2720-11e5-96c5-a5b38f6c42d3) fails.